src/relay/backend/vm/compiler.cc

```diff
@@ -266,6 +266,8 @@ class VMFunctionCompiler : ExprFunctor<void(const Expr& expr)> {
 
   void VisitExpr_(const ConstantNode* const_node) {
     size_t konst_idx = context_->constants.size();
+    std::string name = "p" + std::to_string(konst_idx);
+    context_->constant_indices[name] = konst_idx;
     context_->constants.push_back(const_node->data);
```

Comment:
One question here. Will the parameters be stored in the VM constants as well?

Reply:
Yes, https://github.com/dmlc/tvm/pull/4077/files#diff-55946b9ca058531c94f3c6c7195c802bR903-R905. We save the constants into vm object. We need to store constant shape tensor anyway since the information is not available elsewhere. it's possible we can optimize out the storage for nn weights if we expect user to call `load_params`.